new-frontend/src/app/components/permissions-matrix/permissions-matrix.component.ts

```diff
@@ -0,0 +1,193 @@
+              const operationPermission = new OperationPermission(resource, scope, operation, cellState);
+              if (cellState === CellState.Selected) this.selection.select(operationPermission);
+              return operationPermission;
+            });
+            return new ScopePermission(scope, operationPermissions);
+          });
+
+        const resourcePermission = new ResourcePermission(resource, scopePermissions);
+        return resourcePermission;
+      })
+      .filter((resourcePermission) => resourcePermission.scopes.length > 0);
+
+    this.selectionSubscription = this.selection.changed.subscribe(() => this.handleSelectionChange());
+
+    return result;
+  }
+
+  private hasSelectableOperations(resource: ResourceType, scope: ScopeType, selectable: Rule[]): boolean {
+    return !!selectable.find((rule) => isEqualString(rule.name, resource) && isEqualString(rule.scope, scope));
+  }
+
+  private getCellState(
+    fixedSelected: Rule[],
+    preselected: Rule[],
+    selectable: Rule[],
+    resource: ResourceType,
+    scope: ScopeType,
+    operation: OperationType
+  ): CellState {
+    if (!this.containsRule(selectable, resource, scope, operation)) return CellState.NotApplicable;
+
+    if (this.containsRule(fixedSelected, resource, scope, operation)) return CellState.FixedSelected;
+
+    const isAllowed = this.permissionService.isAllowedToAssignRuleToRole(scope, resource, operation);
+    const isPreselected = this.containsRule(preselected, resource, scope, operation);
+
+    if (isAllowed && isPreselected) return CellState.Selected;
+    if (isAllowed && !isPreselected) return CellState.NotSelected;
+    if (!isAllowed && isPreselected) return CellState.FixedSelected;
+
+    return CellState.FixedNotSelected;
+  }
+
+  private containsRule(rules: Rule[], resource: ResourceType, scope: ScopeType, operation: OperationType): boolean {
+    return !!this.findRule(rules, resource, scope, operation);
+  }
+
+  private findRule(rules: Rule[], resource: ResourceType, scope: ScopeType, operation: OperationType): Rule | undefined {
+    return rules.find(
+      (rule) => isEqualString(rule.name, resource) && isEqualString(rule.scope, scope) && isEqualString(rule.operation, operation)
+    );
+  }
+
+  public showCheckBox(operationPermission: OperationPermission): boolean {
+    return [CellState.Selected, CellState.NotSelected, CellState.FixedNotSelected].includes(operationPermission.state);
+  }
+
+  public isDisabled(operationPermission: OperationPermission): boolean {
+    return operationPermission.state === CellState.FixedNotSelected;
+  }
+
+  public showCheckIcon(operationPermission: OperationPermission): boolean {
+    return operationPermission.state === CellState.FixedSelected;
+  }
+
+  public rowClass(permissionIndex: number): string {
+    return permissionIndex % 2 === 0 ? 'roles-table__even-row' : 'roles-table__uneven-row';
```

Comment:
Just curious: doesn't Angular Material have some default styling to achieve this?